File: programmers/level 2/서버_증설_횟수.py
# 250429
def solution(players, m, k):
    from collections import deque

    q = deque()
    total_cnt = cnt = 0

    for i in range(24):
        player = players[i]
        if player >= m * (cnt + 1):
            new_cnt = player // m
            q.append((i + k - 1, new_cnt - cnt))
            total_cnt += new_cnt - cnt
            cnt = new_cnt

        if q and q[0][0] == i:
            cnt -= q.popleft()[1]

    return total_cnt

File: programmers/level 2/서버_증설_횟수.js
// 250429
function solution(players, m, k) {
    let [totalCnt, cnt, j] = [0, 0, 0]
    const q = []

    players.forEach((player, i) => {
        if (player >= (cnt+1)*m) {
            const newCnt = Math.floor(player/m)
            q.push([i+k-1, newCnt - cnt])
            totalCnt += newCnt - cnt
            cnt = newCnt
        }

        if (q.length > j && q[j][0] === i) {
            cnt -= q[j][1]
            j += 1
        }
    })


    return totalCnt
}